### PR TITLE
Setup/UI structure -- Build and style plugins page

### PR DESCRIPTION
**Pull request by M-AminAlizadeh**
The things that I did in this PR:

- Add one line to the end of each file
- Install and use react-router for routing
- Add favicon
- Build and style plugins page completely
- Add a search feature on the plugins page
- Fetch plugins and display them
- Responsiveness

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -3,9 +3,10 @@
   <head>
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
-    <title>Logsec Marketplace</title>
+    <link rel="icon" href="https://logseq.com/logo-with-border.5bf84f43.png" />
```
Comment (ahangarha) [inline public/index.html:6]: It is fine for now but let's keep in mind that the hash part of the name might get changed in the future.

File: src/components/Plugin.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+const Plugin = ({
+  id, icon, title, author, description, repo,
+}) => {
+  const properIconUrl = (icon) => {
+    if (icon.includes('/')) {
+      return icon.split('/')[1];
+    }
+    return icon;
+  };
```
Comment (ahangarha) [inline src/components/Plugin.jsx:11]: ## ❌ Changes Required

What if the icon has such value: `./img/icon.png`?

I think you meant to remove `./` if the icon path starts with it.

File: src/components/Plugin.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  return (
+    <section className="border-2 p-3 rounded-md">
+      <img
+        width="100px"
+        src={icon ? `https://raw.githubusercontent.com/logseq/marketplace/master/packages/${id}/${properIconUrl(icon)}` : 'https://img.icons8.com/pastel-glyph/64/page-not-found--v2.png'}
```
Comment (ahangarha) [inline src/components/Plugin.jsx:17]: ## 💡 Optional

Maybe it is better to extract this logic into a function like `getIconPath(icon)`.
It helps to keep the JSX part much cleaner.

I personally prefer to use a ternary operation when it is tiny and short.

**Review by ahangarha — CHANGES_REQUESTED** [review]
## ❌ Changes Required

This PR is much better than what I expected. 👍🏾 You have done great work so far. There is a tiny but important bug. Please check the comments.

I also made some other comments for optional changes. It is up to you to do it now or later. Maybe you can combine them while addressing the bug fix.

File: src/components/Plugin.jsx (hunk 3 above)
Comment (M-AminAlizadeh) [reply src/components/Plugin.jsx:17]: Good Idea 
I changed it

File: public/index.html (hunk 1 above)
Comment (M-AminAlizadeh) [reply public/index.html:6]: I changed it from absolute path to relative path.

File: src/components/Plugin.jsx (hunk 2 above)
Comment (M-AminAlizadeh) [reply src/components/Plugin.jsx]: Nice catch 
But according to the values of the icon key in each object, all of them have similar paths like `icon.png` or `./icon.png` or similar and I didn't see any value like `./dir/icon.png` or something like that.
I built this logic in `properIconUrl` according to the plugin objects that are already shared maybe in the future if something changes I will change the logic.
For now, I just changed the logic from `/` to `./`

**Review by ahangarha — APPROVED** [review]
Good work 👍🏾

**Pull request merged**